elasticapm/contrib/starlette/__init__.py

```diff
@@ -192,3 +194,29 @@ async def _request_finished(self, response: Response):
 
         result = "HTTP {}xx".format(response.status_code // 100)
         elasticapm.set_transaction_result(result, override=False)
+
+    def get_route_name(self, request: Request) -> str:
+        route_name = None
+        app = request.app
+        scope = request.scope
+        routes = app.routes
+
+        for route in routes:
+            match, _ = route.matches(scope)
+            if match == Match.FULL:
+                route_name = route.path
+                break
+            elif match == Match.PARTIAL and route_name is None:
+                route_name = route.path
+        # Starlette magically redirects requests if the path matches a route name with a trailing slash
+        # appended or removed. To not spam the transaction names list, we do the same here and put these
+        # redirects all in the same "redirect trailing slashes" transaction name
+        if not route_name and app.router.redirect_slashes and scope["path"] != "/":
+            redirect_scope = dict(scope)
+            redirect_scope["path"] = scope["path"][:-1] if scope["path"].endswith("/") else scope["path"] + "/"
+            for route in routes:
+                match, _ = route.matches(redirect_scope)
+                if match != Match.NONE:
+                    route_name = "redirect trailing slashes"
```

Comment:
```suggestion
                    route_name = "<internal redirect>"
```

What do you think about this? I'm not sure the "trailing slashes" will mean anything to the average user.

I do like collecting all of these transactions in a separate bucket so that they don't affect the stats for the routes to which they redirect, I'm just not sure of the name.

Reply:
I struggled with naming, and you're probably right that my choice isn't super understandable by users not familiar with the internals of Starlette routing (especially because the `Router.redirect_slashes` flag seems to be undocumented and not configurable when instantiating a `Starlette` application). `<internal redirect>` seems to be very broad though, as we won't put any other kinds of redirects into that name bucket...

Reply:
Eh, I don't feel strongly. We can keep it as-is.

Reply:
Love the changes! Can't wait to see this in master.

As a user, I would probably expect the following behaviour:

```suggestion
                    route_name = route.path[:-1]
```

I _think_ that would be somewhat in line with what we see with other frameworks, that expose the route information on the request objects (I could be wrong though).

Reply:
@HenrikOssipoff I implemented your suggestion now. I'm a bit concerned that it could spam the transaction list with lots of these transaction names that are only redirects to real transactions, but I guess that's only a problem if the wrong URLs are used consistently.